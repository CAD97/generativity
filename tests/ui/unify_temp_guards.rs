use generativity::{guard, Guard};

fn unify<'id>(_: Guard<'id>, _: Guard<'id>) {
    unreachable!()
}

fn main() {
    match (guard!(), guard!()) {
        (a, b) => unify(a, b),
    }
}
