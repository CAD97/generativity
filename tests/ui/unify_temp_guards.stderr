error[E0716]: temporary value dropped while borrowed
  --> tests/ui/unify_temp_guards.rs:8:22
   |
8  |     match (guard!(), guard!()) {
   |            --------  ^^^^^^^^ creates a temporary which is freed while still in use
   |            |
   |            a temporary with access to the borrow is created here ...
...
11 | }
   | -
   | |
   | temporary value is freed at the end of this statement
   | ... and the borrow might be used here, when that temporary is dropped and runs the `Drop` code for type `Brand`
   |
   = note: consider using a `let` binding to create a longer lived value
   = note: the temporary is part of an expression at the end of a block;
           consider forcing this temporary to be dropped sooner, before the block's local variables are dropped
   = note: this error originates in the macro `guard` (in Nightly builds, run with -Z macro-backtrace for more info)
help: for example, you could save the expression's value in a new local variable `x` and then make `x` be the expression at the end of the block
   |
19 |         unsafe { let x = $crate::Guard::tie(&$crate::Brand::tie(&$crate::Id::new())); x }
   |                  +++++++                                                            +++
